tools/swift-inspect/Sources/swift-inspect/Process.swift

```diff
@@ -43,6 +43,59 @@ private func refersToSelf(_ str: String) -> Bool {
   // No match.
   return false
 }
+
+internal func getRemoteProcess(processId: ProcessIdentifier,
+                               options: UniversalOptions) -> (any RemoteProcess)? {
+  return DarwinRemoteProcess(processId: processId,
+                             forkCorpse: options.forkCorpse)
+}
+
+internal func getProcessName(processId: ProcessIdentifier) -> String? {
+  var info = proc_bsdinfo()
+  let bsdinfoSize = Int32(MemoryLayout<proc_bsdinfo>.stride)
+  let size = proc_pidinfo(processId, PROC_PIDTBSDINFO, 0, &info, bsdinfoSize)
+  if (size != bsdinfoSize) {
+    return nil
+  }
+  let processName = withUnsafePointer(to: info.pbi_name) {
+    $0.withMemoryRebound(to: CChar.self, capacity: MemoryLayout<CChar>.stride) {
+      String(cString: $0)
+    }
+  }
+  return processName
+}
+
+internal func getAllProcesses(options: UniversalOptions) -> [ProcessIdentifier]? {
+  var ProcessIdentifiers = [ProcessIdentifier]()
+  let kinfo_stride = MemoryLayout<kinfo_proc>.stride
+  var bufferSize: Int = 0
+  var name: [Int32] = [CTL_KERN, KERN_PROC, KERN_PROC_ALL]
+
+  guard sysctl(&name, u_int(name.count), nil, &bufferSize, nil, 0) == 0 else {
+    return nil
+  }
+  var count = bufferSize / kinfo_stride
+  var buffer = Array(repeating: kinfo_proc(), count: count)
+  guard sysctl(&name, u_int(name.count), &buffer, &bufferSize, nil, 0) == 0 else {
+    return nil
+  }
```

Comment:
We should probably shrink the array if `bufferSize` is smaller after the second call (in case a process terminated between the first and second calls). Otherwise we'll loop over the unfilled info structs at the end. It looks like the code below should handle that gracefully, but it would be cleaner to remove them explicitly.